### PR TITLE
fix(nuxi): preview nitro build with custom dir config

### PR DESCRIPTION
**Pull request by xjccc**
### 🔗 Linked issue
#18879 

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation or readme)
- [x] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

when config nitro.output.dir
command nuxi preview is error the path

i have config the path from nitro.output.dir in preview

is this the right way to deal with the problem?

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by danielroe** [comment]
I'm reluctant to merge this as-is because it adds a bit of overhead to the preview command, which is really just wrapper anyway to running a command specified in the output folder's `nitro.json`.

Given that there's already a way to do this (https://github.com/nuxt/nuxt/issues/18879#issuecomment-14222718500) I wonder if this is needed. cc: @pi0

**Comment by xjccc** [comment]
> I'm reluctant to merge this as-is because it adds a bit of overhead to the preview command, which is really just wrapper anyway to running a command specified in the output folder's `nitro.json`.
> 
> Given that there's already a way to do this ([#18879 (comment)](https://github.com/nuxt/nuxt/issues/18879#issuecomment-14222718500)) I wonder if this is needed. cc: @pi0

yes.it has a way to deal the problem.(nuxi preview dir)
it is my simple idea.
i hope deal better.
thank u

**Comment by pi0** [comment]
Preview (same as `dev`) is local. Overhead is probably not much noticeable. However it won't cover all custom presets, it is also not a bad idea to load configuration for preview improving DX of it with custom config support.

File: packages/nuxi/src/commands/preview.ts [hunk 1]
```diff
@@ -16,8 +17,11 @@ export default defineNuxtCommand({
   async invoke (args) {
     process.env.NODE_ENV = process.env.NODE_ENV || 'production'
     const rootDir = resolve(args._[0] || '.')
+    const { loadNuxtConfig } = await loadKit(rootDir)
+    const config = await loadNuxtConfig({ cwd: rootDir })
+    const serverOutputDir = config.nitro.output?.dir?.replace(/^~\//, '') || '.output'
```
Comment (pi0) [inline packages/nuxi/src/commands/preview.ts:22]: Probably not supporting `~` (it is srcDir while config is supposed to be from rootDir/workspaceDir)

**Review by pi0 — APPROVED** [review]

**Review by danielroe — APPROVED** [review]

**Comment by xjccc** [comment]
@danielroe @pi0 
thank you very much. this is my first pr(however i worked less [smile]),
but its very encouraging for me. i hope i can make more contributions in the future.
once again, thank you for your support and appreciate your efforts and hard work.

**Pull request merged**